Tools/peg_generator/pegen/c_generator.py

```diff
@@ -28,6 +28,15 @@
 EXTENSION_PREFIX = """\
 #include "pegen.h"
 
+#define RAISE_SYNTAX_ERROR(msg, ...) _PyPegen_raise_error({parser_var}, PyExc_SyntaxError, 1, msg, ##__VA_ARGS__)
+#define RAISE_INDENTATION_ERROR(msg, ...) _PyPegen_raise_error({parser_var}, PyExc_IndentationError, 1, msg, ##__VA_ARGS__)
+#define RAISE_SYNTAX_ERROR_NO_COL_OFFSET(msg, ...) _PyPegen_raise_error({parser_var}, PyExc_SyntaxError, 0, msg, ##__VA_ARGS__)
+#define UNUSED(expr) do {{ (void)(expr); }} while (0)
+#define EXTRA_EXPR(head, tail) head->lineno, head->col_offset, tail->end_lineno, tail->end_col_offset, {parser_var}->arena
+#define EXTRA __start_lineno, __start_col_offset, __end_lineno, __end_col_offset, {parser_var}->arena
+#define CHECK_VERSION(version, msg, node) INVALID_VERSION_CHECK({parser_var}, version, msg, node)
+#define CHECK(result) CHECK_CALL({parser_var}, result)
+#define CHECK_NULL_ALLOWED(result) CHECK_CALL_NULL_ALLOWED({parser_var}, result)
```

Comment:
I feel that these shouldn't be in `c_generator.py`. If we ever need a new MACRO, it's certainly not the first place that comes to mind for where to define them. Maybe define these in the `header` meta in the grammar file?

Reply:
Yeah, that's a good idea. Originally I placed these here because we needed to substitute the parser variable in them, but if we place them in the grammar is like any other "p" that we put there, which should match the choice of "parser_variable". Will do this soon

Reply:
@lysnikolaou One downside of this that I found is that now we would need to modify all test strings of `test_c_parser` to include the same macros if we go this route.